src/microvm/src/vmm/microvm/io.rs

```diff
@@ -146,45 +127,37 @@ impl IoThread {
         gateway: Gateway,
         microvm_rx: Receiver<Message>,
         microvm_tx: Sender<Message>,
-        control_input_tx: Sender<ControlCommand>,
-        control_output_rx: Receiver<ControlCommandResponse>,
+        vmm_control_tx: Sender<ControlCommand>,
+        vmm_control_rx: Receiver<ControlCommandResponse>,
     ) -> Result<Self> {
         Ok(Self {
             gateway,
             microvm_rx,
             microvm_tx,
             incoming: VecDeque::new(),
             outgoing: VecDeque::new(),
-            _state: OrchestratorState::PreBoot,
-            _control_input_tx: control_input_tx,
-            _control_output_rx: control_output_rx,
+            vmm_control_tx,
+            vmm_control_rx,
         })
     }
 
     ///
     /// # Description
     ///
-    /// Runs the I/O thread.
+    /// Runs the I/O thread according to the state in the snapshotting protocol state machine.
     ///
     /// # Returns
     ///
     /// Upon success, empty is returned. Otherwise, an error is returned instead.
     ///
     fn run(&mut self) -> Result<()> {
-        let mut round: usize = 0;
-
-        // Cycle through actions to avoid starvation.
         loop {
-            if round % 4 == 0 {
-                self.try_receive_from_microvm()?;
-            } else if round % 4 == 1 {
-                self.try_send_to_gateway()?;
-            } else if round % 4 == 2 {
-                self.try_receive_from_gateway()?;
-            } else {
-                self.try_send_to_microvm()?;
-            }
-            round += 1;
+            self.try_receive_from_microvm()?;
+            self.try_send_to_gateway()?;
+            self.try_receive_from_vmm_control()?;
+            self.try_send_to_microvm()?;
+            self.try_receive_from_gateway()?;
+            self.mock_receive_from_outside_control_and_send_to_vmm_control()?;
```

Comment:
I am still not convinced why this is required. If no control commands are sent at all, there should be no impact in performance. If no commands are sent at all, this statement is not required.